### PR TITLE
atomically create the CACHE_TAG file, see #6028

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: src/borg/helpers/fs.py
Comment (ThomasWaldmann) [inline src/borg/helpers/fs.py:103]: Hmm, nope.

That would create, write and rename the tag file for each call of `get_cache_dir()`, which is expensive / bad in other ways.

So guess we still want to use the "not exists" condition, even if slightly racy, and combine it with the code seen above.

Comment (ThomasWaldmann) [reply src/borg/helpers/fs.py]: solved, see commits

**Comment by ThomasWaldmann** [comment]
@enkore @textshell can you review please?

**Comment by ThomasWaldmann** [comment]
BTW, I thought about using a random SUFFIX for the tmp file (tempfile stdlib module style or so).

But guess that would lose the nice .tmp file cleanup property of the current code (as the current code always kills the always-same-suffix .tmp file first, if any is present).

**Comment by hexagonrecursion** [comment]
I can backport this. Should I wait until this is merged?

**Comment by ThomasWaldmann** [comment]
Yes, please wait until merged. Thanks!

**Review by enkore — APPROVED** [review]

**Comment by ThomasWaldmann** [comment]
Thanks @enkore for having a look. Ready for backporting to 1.1-maint now.

**Comment by hexagonrecursion** [comment]
Sorry. Life. Busy. I don't know when I will be able to work on backporting this.

**Pull request merged**
